compiler/haskell-ide-core/src/Development/IDE/State/Shake.hs

```diff
@@ -344,7 +344,11 @@ type instance RuleResult (Q k) = A (RuleResult k)
 -- | Compute the value
 uses :: IdeRule k v
     => k -> [FilePath] -> Action [Maybe v]
-uses key files = map (\(A value _) -> value) <$> apply (map (Q . (key,)) files)
+uses key files = map (\(A value _) -> value) <$> apply (map (Q . (key,)) (normaliseNonEmpty <$> files))
+
+normaliseNonEmpty :: FilePath -> FilePath
+normaliseNonEmpty "" = ""
```

Comment:
Why do we get empty file paths in here?

Reply:
We use empty file paths for rules that are not associated with a file.